examples/proc_GDS_pulse_capture.py

```diff
@@ -12,147 +12,160 @@
+
+        def switch_to_plot(d, j):
+            thislen = d["t_pulse"][j]
+            fl.next(f"pulse length = {thislen}")
+
+        def indiv_plots(d, thislabel, thiscolor):
+            if skip_plots is None:
+                return
+            for j in range(len(d["t_pulse"])):
+                if j % skip_plots == 0:
+                    switch_to_plot(d, j)
+                    fl.plot(
+                        d["t_pulse", j],
+                        alpha=0.2,
+                        color=thiscolor,
+                        label=thislabel,
+                    )
+
+        indiv_plots(d, "raw", "blue")
+        # {{{ make analytic, then filter
+        d.ft("t", shift=True)
+        d = d["t":(0, None)]
+        d.ift("t")
+        d *= 2
+        d["t":0] *= 0.5
+        indiv_plots(abs(d), "analytic", "orange")
+        d.ft("t")
+        d["t":(0, 11e6)] *= 0
+        d["t":(24e6, None)] *= 0
+        d.ift("t")
+        indiv_plots(abs(d), "filtered analytic", "red")
         # }}}
-        for j in range(len(d["p_90"])):
-            s = d["p_90", j].C
-            s.set_units("t", "s")
-            s *= atten_ratio
-            s /= np.sqrt(2)  # Vrms
-            s /= np.sqrt(50)  # V/sqrt(R) = sqrt(P)
-            if show_all:
-                fl.basename = r"pulse length = %s" % str(s.get_prop("set_p90s")[j])
-                fl.next(fl.basename)
-                # {{{ Plot raw capture
-                fl.plot(s, alpha=0.2, color="blue")
-                # }}}
-            # {{{ make analytic
-            s.ft("t", shift=True)
-            s = s["t":(0, None)]
-            s *= 2
-            s["t":0] *= 0.5
-            # }}}
-            if show_all:
-                s.ift("t")
-                # {{{ Plot analytic
-                fl.plot(abs(s), color="orange")
-                s.ft("t")
-                # }}}
-            # {{{ frequency filter
-            s["t":(0, 11e6)] *= 0
-            s["t":(24e6, None)] *= 0
-            # }}}
-            s.ift("t")
+        thislabel = "amplitude = %f" % amplitude
+        thiscolor = next(color_cycle)
+        beta = d.shape.pop("t").alloc(dtype=np.float64)
+        beta.copy_axes(d)
+        beta.set_units(r"μs√W")
+        for j in range(len(d["t_pulse"])):
+            s = d["t_pulse", j]
+            thislen = d["t_pulse"][j]
             int_range = abs(s).contiguous(lambda x: x > 0.01 * s.max())[0]
             # slightly expand int range to include rising edges
             int_range[0] -= 1e-6
             int_range[-1] += 1e-6
-            beta1 = abs(s["t":int_range]).integrate("t").data.item() * 1e6
-            beta_v_t["At_p", j] = beta1
-            if show_all:
-                # {{{ plot and integrate 90 pulse
-                fl.plot(abs(s["t":int_range]), color="red")
-                # }}}
+            beta["t_pulse", j] = (
+                abs(s["t":int_range]).integrate("t").data.item() * 1e6
+            )
```

Comment:
note that this does not make sense, which again makes me suspicious that your axis labels are incorrect.

Reply:
I think we are on the same page here?